cql3/expr/restrictions.cc

```diff
@@ -22,28 +22,6 @@ bool is_legal_relation_for_non_frozen_collection(oper_t oper, bool is_lhs_col_in
     return oper == oper_t::CONTAINS_KEY || oper == oper_t::CONTAINS || (oper == oper_t::EQ && is_lhs_col_indexed);
 }
 
-bool oper_is_slice(oper_t oper) {
-    switch(oper) {
-        case oper_t::LT:
-        case oper_t::LTE:
-        case oper_t::GT:
-        case oper_t::GTE:
-            return true;
-        case oper_t::EQ:
-        case oper_t::EQ_SPECIAL_NULL:
-        case oper_t::NEQ:
-        case oper_t::NEQ_SPECIAL_NULL:
-        case oper_t::IN:
-        case oper_t::IN_SPECIAL_NULL:
-        case oper_t::CONTAINS:
-        case oper_t::CONTAINS_KEY:
-        case oper_t::IS_NOT:
-        case oper_t::LIKE:
-            return false;
-    }
-    on_internal_error(expr_logger, format("oper_is_slice: Unhandled oper_t: {}", oper));
-}
-
```

Comment:
I didn't understand how this patch can remove all this code. Maybe you can explain this in the commit message?

Reply:
I'll update the commit message. Preview: it duplicates is_slice() from expression.hh, and the expression code does all these checks.